api/src/main/java/com/basistech/rosette/api/RosetteAPI.java

```diff
@@ -1753,6 +1758,12 @@ private String headerValueOrNull(Header header) {
             throws IOException, RosetteAPIException {
         int status = httpResponse.getStatusLine().getStatusCode();
         String encoding = headerValueOrNull(httpResponse.getFirstHeader(HttpHeaders.CONTENT_ENCODING));
+        try {
+            int concurrencyHeader = Integer.parseInt(headerValueOrNull(httpResponse.getFirstHeader("X-RosetteApi-Concurrency")));
```

Comment:
Yeah, don't trust the server will always do the right thing ...
Also no need to close PR to make changes, you can keep pushing and changes will show in the same PR.